### PR TITLE
172 youtube

### PR DESCRIPTION
**Pull request by Toni-Zoaretz**

### CODE REVIEW

**Review by yishain11 — CHANGES_REQUESTED** [review]
1. please make sure that all the imports work (youtube file with lower case for example)
2. add serpi keys to the generate config script in server
3. please add tests for the new functions/endpoints

File: server/controllers/songsController.js [hunk 1]
```diff
@@ -127,37 +129,70 @@ const getFullSongData = asyncHandler(async (req, res, next) => {
       // if not - generate song data - > save song in db
       logger.info(`no songs found for song name: ${song}.generating data`);
       const songData = await generateSongData(song, artist, coverArt);
+      // console.log("SONG DATA ARE", songData);
+      //
+      // const songYoutubeId = await generateYoutubeId(song, artist);
+      //
```
Comment (yishain11) [inline server/controllers/songsController.js:135]: remove

File: server/controllers/songsController.js [hunk 2]
```diff
@@ -127,37 +129,70 @@ const getFullSongData = asyncHandler(async (req, res, next) => {
...
+  );
+
+  try {
+    const names3langs = await translateText3Lang(song);
+    finalSongData = {
+      ...finalSongData,
+      ...{ name: names3langs },
+    };
+  } catch (error) {
+    console.log("tran text 3 ", error.response.data.error);
```
Comment (yishain11) [inline server/controllers/songsController.js:166]: write meanigfull msg

File: server/controllers/songsController.js [hunk 3]
```diff
@@ -127,37 +129,70 @@ const getFullSongData = asyncHandler(async (req, res, next) => {
...
+  } catch (error) {
+    console.log("prapare artist", error.response.data.error);
+  }
+
+  const songYoutubeId = generateYoutubeId(song, artist)
+    .then((data) => {
+      finalSongData.youtubeURL = data;
+    })
+    .catch((err) => {
+      console.log("error is", err);
```
Comment (yishain11) [inline server/controllers/songsController.js:181]: write meanigfull msg

File: server/controllers/songsController.js [hunk 4]
```diff
@@ -127,37 +129,70 @@ const getFullSongData = asyncHandler(async (req, res, next) => {
...
+  try {
+    // get lyrics
+    const { lyricsObj, originalLang } = await prepareLyrics(song, artist);
+    finalSongData = {
+      ...finalSongData,
+      ...{ lyrics: lyricsObj },
+      ...{ originalLang },
+    };
+  } catch (error) {
+    console.log("prepare", error);
```
Comment (yishain11) [inline server/controllers/songsController.js:193]: write meanigfull msg

File: server/tests/generateSongData.tests.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+    const data = response.data;
+    expect(response.status).to.be.equal(200);
+    expect(data).to.be.a("object");
+    expect(data.name.hebrew).to.be.a("string").and.not.empty;
+    expect(data.name.arabic).to.be.a("string").and.not.empty;
+    expect(data.name.english).to.be.a("string").and.not.empty;
+    expect(data.lyrics.hebrew).to.be.a("string").and.not.empty;
+    expect(data.lyrics.arabic).to.be.a("string").and.not.empty;
+    expect(data.lyrics.english).to.be.a("string").and.not.empty;
```
Comment (yishain11) [inline server/tests/generateSongData.tests.js:30]: good for checking many things, but where is the testing for the main thing - youtube id?

**Review by yishain11 — CHANGES_REQUESTED** [review]

File: client/src/components/Lyrics/Lyrics.jsx [hunk 6]
```diff
@@ -45,7 +51,8 @@ const Lyrics = ({ lyrics, name, originalLang }) => {
               <Paragraph>{originalLyrics}</Paragraph>
             )}
           </div>
-        </div>
+          ;
```
Comment (yishain11) [inline client/src/components/Lyrics/Lyrics.jsx:54]: remove ;

File: client/src/components/Lyrics/LyricsStyle.jsx [hunk 7]
```diff
@@ -40,11 +38,12 @@ export const Paragraph = styled.p`
...
   font-size: 0.7rem;
-  margin-bottom: 1.5rem;
   color: black;
   width: 16ch;
-  height: 5ch;
   white-space: pre-line;
 `;
+
+export const WrapperDiv = styled.div`
+  height: 100%;
```
Comment (yishain11) [inline client/src/components/Lyrics/LyricsStyle.jsx:48]: if this is the only style applied, add it inline in jsx, no need to create a styled component for this

File: client/src/components/Youtube/Youtube.jsx [hunk 8]
```diff
@@ -0,0 +1,20 @@
+import { YoutubeBox } from "./YoutubeStyle";
+
+function Youtube({ songName, artistName, youtubeUrl }) {
+  return (
+    <div>
+      <YoutubeBox>
+        <iframe
+          width="560"
```
Comment (yishain11) [inline client/src/components/Youtube/Youtube.jsx:8]: I think the numbers should be dynamic

File: server/tests/generateSongData.tests.js [hunk 9]
```diff
@@ -0,0 +1,32 @@
+import chai from "chai";
+import axios from "axios";
+
+const expect = chai.expect;
+
+describe.only("Test suite get full song data", function () {
```
Comment (yishain11) [inline server/tests/generateSongData.tests.js:6]: remove .only

**Review by yishain11 — CHANGES_REQUESTED** [review]

File: client/src/components/Youtube/Youtube.jsx (hunk 8 above)
Comment (Toni-Zoaretz) [reply client/src/components/Youtube/Youtube.jsx:8]: no its should be the same for all the videos i check this with a few videos

**Pull request merged**